File: golang-for-nodejs-developers/examples/objects.js
const obj = {
  someProperties: {
    foo: 'bar'
  },
  someMethod(prop) {
    return this.someProperties[prop]
  }
}

let item = obj.someProperties.foo
console.log(item)

item = obj.someMethod('foo')
console.log(item)

File: golang-for-nodejs-developers/examples/objects.go
package main

import "fmt"

type Obj struct {
	SomeProperties map[string]any
}

func NewObj() *Obj {
	return &Obj{
		SomeProperties: map[string]any{
			"foo": "bar",
		},
	}
}

func (o *Obj) SomeMethod(prop string) any {
	return o.SomeProperties[prop]
}

func main() {
	obj := NewObj()

	item := obj.SomeProperties["foo"]
	fmt.Println(item)

	item = obj.SomeMethod("foo")
	fmt.Println(item)
}
